Source/CarthageKit/Xcode.swift

```diff
@@ -735,14 +811,64 @@ private func resolveSameTargetName(for settings: BuildSettings) -> SignalProduce
 	}
 }
 
+/// Using target architecture information in `settings`, copy platform-specific framework bundles from all
+/// xcframeworks in `buildDirectory` to a temporary directory.
+///
+/// The extracted frameworks are used to support building projects which are not configured to use XCFramework products
+/// from their Carthage/Build directory.
+///
+/// Sends the temporary directory or `nil` if there are no xcframeworks to extract.
+func extractXCFrameworks(in buildDirectory: URL, for settings: BuildSettings) -> SignalProducer<URL?, CarthageError> {
+	guard let platformTripleOS = settings.platformTripleOS.value else {
+		// Can't extract xcframeworks if this project doesn't declare its OS triple
+		return SignalProducer(value: nil)
+	}
+
+	let findFrameworks = SignalProducer<[URL]?, CarthageError> {
+		try? FileManager.default.contentsOfDirectory(at: buildDirectory.resolvingSymlinksInPath(), includingPropertiesForKeys: nil)
+	}
+	.skipNil()
+	.flatten()
+	.filter { $0.pathExtension == "xcframework" }
+	.flatMap(.merge) { url -> SignalProducer<URL, CarthageError> in
+		frameworkBundlesInURL(url, compatibleWith: platformTripleOS, variant: settings.platformTripleVariant.value)
+			.mapError { .readFailed(url, $0 as NSError) }
+			.map { $0.bundleURL }
+	}
+
+	let makeTemporaryDirectory = SignalProducer<URL, CarthageError> { () -> Result<URL, CarthageError> in
+		var templatePath = (NSTemporaryDirectory() as NSString).appendingPathComponent("carthage-xcframework-XXXX").utf8CString
```

Comment:
is this directory ever deleted?

Reply:
It is now: https://github.com/Carthage/Carthage/pull/3071/commits/f151e9a14e6b2361ef78f905eec15b1898f631ee